File: react-app/src/types/HttpTypes/Output/RoomJoinOutput.ts
import { ChatMessage } from "../../ChatMessage";
import { PlaylistVideo } from "../../PlaylistVideo";
import { UserPermissions } from "../../UserPermissions";
import { User } from "../../User";
import { VideoPlayerState } from "../../VideoPlayerState";
import { RoomSettings } from "../../RoomSettings";

export type RoomJoinOutput = {
  //User fields
  authorizationToken: string,
  isAdmin: boolean,
  //Room fields
  chatMessages: ChatMessage[],
  playlistVideos: PlaylistVideo[],
  users: User[],
  roomSettings: RoomSettings,
  userPermissions: UserPermissions,
  videoPlayerState: VideoPlayerState
}

File: dotnet-server/Api/HttpClasses/Output/RoomJoinOutput.cs
public class RoomJoinOutput
{
    public string AuthorizationToken { get; set; }
    public bool IsAdmin { get; set; }

    public List<ChatMessage> ChatMessages { get; set; }
    public List<PlaylistVideo> PlaylistVideos { get; set; }
    public List<UserDTO> Users { get; set; }
    public RoomSettings RoomSettings { get; set; }
    public UserPermissions UserPermissions { get; set; }
    public VideoPlayerState VideoPlayerState { get; set; }
}